token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyService.java

```diff
@@ -74,8 +74,11 @@ public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, Map<String, S
 				}
 
 				LOGGER.debug("Successfully retrieved token keys from {} with params {}.", tokenKeysEndpointUri, params);
+				// This is required for app2service communication. In case x-client_cert is provided in the request,
+				// the response can contain identity service broker plan header. Values
 				if (response.containsHeader(X_OSB_PLAN)) {
-					SecurityContext.setServicePlan(response.getFirstHeader(X_OSB_PLAN).getValue());
+					SecurityContext.setServicePlan(
+							response.getFirstHeader(X_OSB_PLAN).getValue().trim().split("\\s*,\\s*"));
```

Comment:
Maybe it would be better to move the parsing to setServicePlan, so the service plan logic is mostly there and the parsing does not have to be re-implemented if this OAuth2TokenKeyService is overridden?